public/js/plantPlacement.js

```diff
@@ -1,86 +1,188 @@
-function checkPlants(modId, x, y) {
-  let article = document.querySelector('#modMap');
-  console.log(article.dataset);
-  let data = JSON.parse(article.dataset.plantedplants)
-  let len = data.length;
-  for (i = 0; i < len; i++) {
-    if (data[i] && data[i].hasOwnProperty("module")) {
-      if (data[i]["module"] == modId) {
-        if (data[i]["y"] == y && data[i]["x"] == x) {
-          //console.log(data[i]["y"]);
-          return data[i]
+
+let selectors = [];
+let plantMatrix = [];
+let uniquePlants = [];
+let data = {};
+
+let article = document.querySelector('#circle_1');
+
+try {
+  data = JSON.parse(article.dataset.plantedplants);
+}
+catch(e) {
+}
+let len = data.length;
+
+function defineUniquePlants() {
+  $.each(data, (index, value) => {
+      if ($.inArray(value.plant, uniquePlants) === -1) {
+          uniquePlants.push(value.plant);
+      }
+  });
+}
+
+function updateSelector(selectorName,plantId) {
+  $(selectorName).val(plantId);
+}
+function checkPlants(x, y) {
+  for (var k = 0; k < len; k++) {
+    if (data[k] ) {
+        if (data[k].y == y && data[k].x == x) {
+          val = data[k].plant;
+          console.log(selectors);
+          a = uniquePlants.indexOf(data[k].plant.toString());
```

Comment:
where's this initialized? also, highly recommend naming variables something human-readable and contextual